core-api/src/main/java/com/optimizely/ab/config/audience/UserAttribute.java

```diff
@@ -118,6 +132,13 @@ public Boolean evaluate(ProjectConfig config, Map<String, ?> attributes) {
         return null;
     }
 
+    private boolean isValidType(String type) {
+        if (type.equals(CUSTOM_ATTRIBUTE.toString()) || type.equals(THIRD_PARTY_DIMENSION.toString())) {
```

Comment:
this is not scalable. adding more types will result in more explicit conditions in this check. may be add the types to a list and check for existence?